worker/codalabworker/run_manager.py

```diff
@@ -0,0 +1,302 @@
+        """
+        Creates a new run which when started will execute the provided run bundle
+        :param bundle: the run bundle to execute
+        :param bundle_path: path on the filesystem where the bundles data is stored
+        :param resources: the resources requested for this run bundle, e.g. cpu, memory, docker image
+        :return: a new Run which will execute the provided run bundle
+        """
+        raise NotImplementedError
+
+    def serialize(self, run):
+        """
+        Serialize the run in order to persist its state.
+        This is used so that workers can pickup where they left off if they are killed.
+        :param run: the run to serialize
+        :return: a dict of the serialized data for the run
+        """
+        raise NotImplementedError
+
+    def deserialize(self, run_data):
+        """
+        Deserialize run data into a run instance.
+        It is expected that the data is from a call to RunManager.serialize(run)
+        :param run_data: the serialized run data
+        :return: a new run instance which was represented by the data
+        """
+        raise NotImplementedError
+
+    def worker_did_start(self):
+        """Provides hook which can be overwritten to do something when the worker starts"""
+        pass
+
+    def worker_will_stop(self):
+        """Provides hook which can be overwritten to do something when the worker will be stopping"""
+        pass
+
+
+class RunBase(object):
+    """
+    Base class for classes which represent an executable run bundle.
+    These are returned from a RunManager and common methods for manipulating the run.
+    """
+
+    @property
+    def bundle(self):
+        raise NotImplementedError
+
+    @property
+    def resources(self):
+        raise NotImplementedError
+
+    @property
+    def dependency_paths(self):
+        """
+        :return: A list of filesystem paths to all dependencies.
+        """
+        return set([dep['child_path'] for dep in self.bundle['dependencies']])
+
+    @property
+    def bundle_path(self):
+        """
+        :return: The filesystem path to the bundle.
+        """
+        raise NotImplementedError
+
+    @property
+    def requested_memory_bytes(self):
+        """
+        If request_memory is defined, then return that.
+        Otherwise, this run's memory usage does not get checked, so return inf.
+        """
+        return self.resources.get('request_memory') or float('inf')
+
+    def start(self):
+        """
+        Start this run asynchronously.
+        :return: True if the run was started, False otherwise.
+        """
+        raise NotImplementedError
+
+    def resume(self):
+        """
+        Resume this run asynchronously.
+        This is used primarily after a worker has deserialized a saved run and then wants to continue it.
+        :return: True if the run could be resumed, False otherwise
+        """
+        raise NotImplementedError
+
+    def kill(self):
```

Comment:
Looks like the `Run` class is still calling this internally and passing a message to it in `_check_and_report_resource_utilization`, so seems like we should add this arg back in. (Unless there's a reason I'm missing that it was removed.)